### ISSUE TITLE
Re-open Support for ALC299

### ISSUE THREAD

**Issue by cepxuo** [insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12#15]
#4 Re-open Support for ALC299
Could you please add ALC299 codec? Here is my codec dump:

[dump.txt](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2097260/dump.txt)

**Comment by cepxuo** [comment]
Hi! the OSX version is 10.4 beta 5 (Mojave)
Thanks!!!

**Comment by cepxuo** [comment]
I am still hoping to find solution. A Im on OSX 10.14 Mojave (not beta) now.

**Comment by insanelydeepak** [comment]
great 
let me fix this soon 
do you have highSierra or Sierra ? will be better for testing

**Comment by cepxuo** [comment]
Unfortunately I have only Mojave available.

**Comment by insanelydeepak** [comment]
send me AppleHDA kext

**Comment by cepxuo** [comment]
Please find it attached:
[AppleHDA.kext.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2595112/AppleHDA.kext.zip)

**Comment by insanelydeepak** [comment]
provide me codec_dump from linux

**Comment by cepxuo** [comment]
And here are codec dumps:
[dumps.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2595395/dumps.zip)

**Comment by insanelydeepak** [comment]
try this use layout13

[AppleHDA-282.10-ALC299.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2595835/AppleHDA-282.10-ALC299.zip)

**Comment by cepxuo** [comment]
Thanks a lot! Audio output works fine. I tried built-in speakers and bluetooth headphones. Now I have only 1 problem with built-in microphone which delivers only noise. The bluetooth microphone works fine.
Could you pleas advise me how to fix built-in microphone?
And thanks again for the patch! It look like you are the only one who could patch AppleHDA for ALC299 codec!!!

**Comment by insanelydeepak** [comment]
wait this is not final version , we must do some more test and then you can use it as final version

**Comment by insanelydeepak** [comment]
Here are details i need : 
1.Speaker is working ? 
2.Mic(internal) is working ?
3.Mic (External) is working 
4. Headphone is working ?

use try headset or phone to check headphone jack is working or not 

also send me screenshot of system=>preferences = > input and output devices

**Comment by cepxuo** [comment]
Oops, I was too  fast with conclusions. The audio device which worked was USB Audio from dock station. When I unplugged it I have no audio devices in System => Preferences => Audio
[Archive.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2599598/Archive.zip)

Attached is the screensots of preferences and screenshot of clover configuration where I configured Audio_ID

**Comment by insanelydeepak** [comment]
arrange these files 
https://github.com/insanelydeepak/cloverHDA-for-Mac-OS-Sierra-10.12/blob/master/TroubleShoot/Troubleshoot%20Reporting.md

**Comment by cepxuo** [comment]
As for steps 4 and 5, I cannot type commands in terminal window of Clover => boot.log. It only allows me to generate log and copy it to a clipboard or file. So I attached the file in the archive with the rest files. The OSX version is 10.14 Mojave with the most recent updates (not beta).
[AppleHDAdebug.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2600733/AppleHDAdebug.zip)

**Comment by insanelydeepak** [comment]
try to get 5 is required like below 

![screen shot 2018-11-20 at 11 00 05 pm](https://user-images.githubusercontent.com/9814566/48791602-15783e80-ed18-11e8-9e19-98e7c86f48e2.png)

**Comment by cepxuo** [comment]
```
kextstat | grep -y AppleHDA 
  137    0 0xffffff7f82486000 0x1d000    0x1d000    com.apple.driver.AppleHDAController (282.10) 1E3D72E3-0918-39C4-B0E7-4561D7915236 <136 45 19 12 7 6 5 4 3 1>
```

**Comment by insanelydeepak** [comment]
sorry i forgot 

please use layout 113 , layout 13 is not supported in Mojave

**Comment by cepxuo** [comment]
```
kextstat | grep -y AppleHDA
  135    0 0xffffff7f82486000 0x1d000    0x1d000    com.apple.driver.AppleHDAController (282.10) 1E3D72E3-0918-39C4-B0E7-4561D7915236 <134 45 19 12 7 6 5 4 3 1>
```
Looks like no changes

**Comment by cepxuo** [comment]
I tried to disable "Reset HDA" checkbox, but still no audio device. However the kextstart log has changed in some values.
```
kextstat | grep -y AppleHDA
  133    0 0xffffff7f82486000 0x1d000    0x1d000    com.apple.driver.AppleHDAController (282.10) 1E3D72E3-0918-39C4-B0E7-4561D7915236 <132 45 19 12 7 6 5 4 3 1>
```

**Comment by insanelydeepak** [comment]
use below commands in terminal 
`sudo rm -f /System/Library/Caches/com.apple.kext.caches/Startup/kernelcache 
sudo touch /System/Library/Extensions
sudo kextcache -i / 
sudo nvram -c
sudo purge
sleep 1`

**Comment by cepxuo** [comment]
I use Kext Utility to install kexts, and it usually clears the cache. I also tried your commands and "sudo nvram -c" exited with following error:
```
sudo nvram -c
nvram: Error clearing firmware variables: (iokit/common) not permitted
```

**Comment by insanelydeepak** [comment]
just use this 

`sudo kextcache -i / `

**Comment by cepxuo** [comment]
Hi! Is there a chance to make it work? As I wrote before, cleaning the cache did not help.

**Comment by insanelydeepak** [comment]
> Unfortunately I need exactly Mojave :( Does it mean we stop trying? Maybe we could try to patch for different layouts?

no , just install 10.13 we will make audio patch and this patch can be use for all OSX versions

**Comment by cepxuo** [comment]
OK, I will find external HDD to install there 10.13 and return back to you. Thanks for patience!!!

**Comment by insanelydeepak** [comment]
bcoz i'm not much familiar with 10.14  yet 10.13 is my daily driver 
send your clover folder 

do you have anyother audio kext installed ?? 

send me IORegistery too

**Comment by cepxuo** [comment]
Please find them attached:
[Archive.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2610643/Archive.zip)

I used to install VoodooHDA before, but I have removed this kext.

**Comment by cepxuo** [comment]
1 step works fine. Your patched sonfig.plist and my old DSDT. Here is the output in terminal:
```
kextstat | grep -y AppleHDA
  141    1 0xffffff7f82497000 0x1d000    0x1d000    com.apple.driver.AppleHDAController (282.10) 1E3D72E3-0918-39C4-B0E7-4561D7915236 <140 45 19 12 7 6 5 4 3 1>
  159    0 0xffffff7f83a8e000 0xb7000    0xb7000    com.apple.driver.AppleHDA (282.10) 88DA0A6E-0D1C-3986-8C95-48A020DD5F77 <158 141 140 119 45 19 6 5 4 3 1>
  165    0 0xffffff7f83698000 0x2000     0x2000     com.apple.driver.AppleHDAHardwareConfigDriver (282.10) 8BB2C14E-F2F9-30FF-B635-9A2C31ACEC67 <4 3>
```
I checked internal speakers and microphone => both work fine.
What did you change in config.plist?

**Comment by insanelydeepak** [comment]
just add few patches like for DSDT know as IRQ Patch 

it is neccessary to load appleHDA 

clover patches => fixTMR, Fix HPET etc 

> I checked internal speakers and microphone => both work fine.

are you sure no USB this time ?

**Comment by cepxuo** [comment]
As you can see:
![screen shot 2018-11-23 at 18 15 32](https://user-images.githubusercontent.com/8061075/48945373-e34f2280-ef4b-11e8-87ae-c929aca2c759.png)
![screen shot 2018-11-23 at 18 15 48](https://user-images.githubusercontent.com/8061075/48945375-e34f2280-ef4b-11e8-8e4c-617ba6de69a7.png)

**Comment by cepxuo** [comment]
Do I need to test anything else? Bluetooth and USB Audio also work fine. Thank you so much!!!

**Comment by cepxuo** [comment]
OK, I have unplugged other sources and here is the result:
1. works fine
2. works fine
3. system still uses internal microphone even though I connect external
4. works fine

**Comment by insanelydeepak** [comment]
> system still uses internal microphone even though I connect external

do you have combojack or just two seperate jack for mic and headphones ??

**Comment by insanelydeepak** [comment]
let me fix your combojack 

question : do you have seperate mic(external) jack too ?

**Comment by insanelydeepak** [comment]
Hi 
try this 
[ALC299.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2612843/ALC299.zip)

**Comment by cepxuo** [comment]
Thank you for help! V4 works fine. This is the only version that brings Line-In available. So now all inputs and all outputs work fine. Unfortunately I don't have separate external microphone, but it works fine with kit headphones from Samsung mobile phone.

**Comment by insanelydeepak** [comment]
> it works fine with kit headphones from Samsung mobile phone.

this is combojack it's working enough

**Comment by insanelydeepak** [comment]
try this one is final and make sure when you plug headphone 
internal speaker switch to headphone same with Mic switched to Line-In 



[AppleHDA-282.10-ALC299 v5.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2613072/AppleHDA-282.10-ALC299.v5.zip)

**Comment by insanelydeepak** [comment]
compare it with v4 and see which doing best result

**Comment by cepxuo** [comment]
V5 does not show any available audio devices (input or output), even though kext seems to be loaded:
```
kextstat | grep -y AppleHDA
  143    1 0xffffff7f82497000 0x1d000    0x1d000    com.apple.driver.AppleHDAController (282.10) 1E3D72E3-0918-39C4-B0E7-4561D7915236 <142 45 19 12 7 6 5 4 3 1>
  149    0 0xffffff7f8261a000 0xb7000    0xb7000    com.apple.driver.AppleHDA (282.10) 88DA0A6E-0D1C-3986-8C95-48A020DD5F77 <148 143 142 111 45 19 6 5 4 3 1>
  156    0 0xffffff7f824b6000 0x2000     0x2000     com.apple.driver.AppleHDAHardwareConfigDriver (282.10) 8BB2C14E-F2F9-30FF-B635-9A2C31ACEC67 <4 3>
```

**Comment by cepxuo** [comment]
> v4 test line-in stay or shows when headphone plug in ??

It shows all the time even though the headphone is not connected.

**Comment by insanelydeepak** [comment]
try this 
just check Line-In switch as headphone and all input/output work
[ALC299.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2613264/ALC299.zip)

**Comment by cepxuo** [comment]
Unfortunately none of 3 shows Line-In and none switches from built-in microphone to Line-In when I plug in the headphones. But all 3 handle output switching fine.

**Comment by insanelydeepak** [comment]
ok 
considering v4 as final version 

also if you want i can fix this too like output but it will consume few more test and all

**Comment by insanelydeepak** [comment]
run this and send me output file
[223866_3074e66e133fb4ecfa4c79c1125f4729.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2615495/223866_3074e66e133fb4ecfa4c79c1125f4729.zip)

**Comment by cepxuo** [comment]
```
odec-Info.command 
HDEF@1F,3 = = = = = = = = = = = = = = = = = = = = = = = = = = = =
Layout, hex: 0x00000071 , dec: 113

PinConfigurations:
0000a6990000004000001790f0111141f0111141f011114100108100f0111141f011114101006040f011114100102100

Codecs Found: 1
HDA:
0 CodecAddress:0x0
VendorID: 0x10ec0299
RevisionID: 0x100002
CodecID: 283902617
Revision(dec):=1048578
Id=0x0299
Id(dec)=665
Vendor=0x10ec
Vendor(dec)=4332

File Doesn't Exist, Will Create: /Users/cepxuo/Desktop/Info.plist
Merge: Error Reading File: Patch.plist
Merge: Error Reading File: Patch.plist
Merge: Error Reading File: Patch.plist
Merge: Error Reading File: Patch.plist
Merge: Error Reading File: Patch.plist
Merge: Error Reading File: Patch.plist
Merge: Error Reading File: Patch.plist
```
and the file:
[Info.plist.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2615840/Info.plist.zip)

**Comment by insanelydeepak** [comment]
i made a test AppleALC kext , after successfull test  i will submit patches to it repo 
copy paste these files in clover=>kexts=>10.14

make sure you have vanilla applehda installed not patched one 
also use layout13 



[Archive.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2616146/Archive.zip)

**Comment by insanelydeepak** [comment]
use 
`kextstat | grep -y alc`

**Comment by cepxuo** [comment]
```
kextstat | grep -y alc
   50    0 0xffffff7f83b80000 0xf0000    0xf0000    as.vit9696.AppleALC (1.3.4) 6FF106C8-E191-31C8-895F-29CA6319494E <43 12 7 5 4 3 2 1>
```

**Comment by insanelydeepak** [comment]
send me config.plist

**Comment by cepxuo** [comment]
[config.plist.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2618726/config.plist.zip)
here it is

**Comment by insanelydeepak** [comment]
use this now 
[files .zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2618737/files.zip)

**Comment by cepxuo** [comment]
Now I have input and output devices, but I don't have Line-In like in V4, and Line-In does not appear when I plug in the headphones. Output is OK.

**Comment by insanelydeepak** [comment]
great 
actually this wasn't test for line-in fix, it was for AppleALC so you can use this kext on all os version without installing patched AppleHDA

**Comment by insanelydeepak** [comment]
try this one 
use install.sh and reboot is must 
[files.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2624328/files.zip)

**Comment by cepxuo** [comment]
Thanks for files! I tried all 5 versions:
V1 and V5 do not have line-in with or without headphones connected.
V3 switches to line-in when I connect headphones, and switches the output to headphones, but in reality headphones are silent and built-in speakers are silent with headphones plugged in.
V2 and V4 seems good, all switching is working, Line-in is switched like output does. All seems to be fine except the volume of headphones' microphone. It is very low, comparing with built-in microphone. I tried to switch adjustment to maximum and tried to switch off the noise detection, but still the volume is very low, even Siri cannot understand what I say. I tried 2 headphones (Samsung and Apple) with the same result (Apple has even more lower volume).

**Comment by cepxuo** [comment]
Unfortunately, I will be unavailable for testing till next Tuesday (December 4), but when I am back I am ready to continue testing! Thank you for your help!!!

**Comment by insanelydeepak** [comment]
sure take your time 

> V1 and V5 do not have line-in with or without headphones connected.
> V3 switches to line-in when I connect headphones, and switches the output to headphones, but in reality headphones are silent and built-in speakers are silent with headphones plugged in.
> V2 and V4 seems good, all switching is working, Line-in is switched like output does. All seems to be fine except the volume of headphones' microphone. It is very low, comparing with built-in microphone. I tried to switch adjustment to maximum and tried to switch off the noise detection, but still the volume is very low, even Siri cannot understand what I say. I tried 2 headphones (Samsung and Apple) with the same result (Apple has even more lower volume).

i can boost volume too will do in next version

**Comment by insanelydeepak** [comment]
> Unfortunately, I will be unavailable for testing till next Tuesday (December 4), but when I am back I am ready to continue testing!

ping me when you available just last final step is left

**Comment by hatf0** [comment]
I can test, I have a XPS 13 with the ALC299 codec. Let me try installing and reporting back my results.

**Comment by hatf0** [comment]
So I've downloaded the v4 patch + the patched AppleALC. Microphone works, however, speakers do not. 
[codec0.txt](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2631740/codec0.txt)

[config.txt](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2631751/config.txt)

**Comment by insanelydeepak** [comment]
Hi 
what's your OSX version ??

**Comment by hatf0** [comment]
10.14.1, and the ALC256 kext patches are a misnomer, I just haven't bothered to rename them. They're disabled, though. 
[Archive.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2635601/Archive.zip)

**Comment by hatf0** [comment]
Codec-Info.command:
```
HDEF@1F,3 = = = = = = = = = = = = = = = = = = = = = = = = =
Layout, hex: 0x0000000d , dec: 13

PinConfigurations:


Codecs Found: 1
HDA:
0 CodecAddress:0x0
VendorID: 0x10ec0299
RevisionID: 0x100002
CodecID: 283902617
Revision(dec):=1048578
Id=0x0299
Id(dec)=665
Vendor=0x10ec
Vendor(dec)=4332

File Doesn't Exist, Will Create: /Users/hatf0/Desktop/Info.plist
Merge: Error Reading File: Patch.plist
Merge: Error Reading File: Patch.plist
Merge: Error Reading File: Patch.plist
Merge: Error Reading File: Patch.plist
Merge: Error Reading File: Patch.plist
Merge: Error Reading File: Patch.plist
Merge: Error Reading File: Patch.plist
```

**Comment by hatf0** [comment]
[card0-codec#0.txt](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2635730/card0-codec.0.txt)
[card0-codec#2.txt](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2635731/card0-codec.2.txt)
All files within CodecDump/ folder

**Comment by insanelydeepak** [comment]
install this , use layout 14 


[AppleHDA-282.10-ALC299_v2.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2636108/AppleHDA-282.10-ALC299_v2.zip)

**Comment by hatf0** [comment]
Installed it, removed AppleALC, and I don't have any audio. Plugging in headphones doesn't get detected either (I have a combojack), but my microphone works. The device does show up, though. I tried installing CodecCommander as well, but that didn't help.

**Comment by hatf0** [comment]
[hackintosh-pro-ioreg.txt](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2636610/hackintosh-pro-ioreg.txt)
[kextstat.txt](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2636611/kextstat.txt)

**Comment by hatf0** [comment]
<img width="593" alt="image" src="https://user-images.githubusercontent.com/15710074/49336158-a4b00c80-f5ca-11e8-88d2-c8feea6d46e1.png">
<img width="589" alt="image" src="https://user-images.githubusercontent.com/15710074/49336160-ac6fb100-f5ca-11e8-9428-e6ee48e3936a.png">

**Comment by insanelydeepak** [comment]
system preferances not system information , check it in Applications 
like below 
![screen shot 2018-12-02 at 11 14 35 am](https://user-images.githubusercontent.com/9814566/49336196-904c2e00-f623-11e8-8f14-62ce26bb0f55.png)

**Comment by insanelydeepak** [comment]
try this 
layout 14

[AppleHDA-282.10-ALC299_v2.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2636628/AppleHDA-282.10-ALC299_v2.zip)

**Comment by insanelydeepak** [comment]
what about headphones ??

**Comment by insanelydeepak** [comment]
try this 
[ALC299.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2636719/ALC299.zip)

**Comment by hatf0** [comment]
So V2 does detect my headphones, but it sounds horrible. It has this high-pitched buzz in the background no matter what, and music sounds like it's being played through a chain of tuna cans over a dialup connection. Speakers do not work, though.

**Comment by insanelydeepak** [comment]
i see 
let me fix speaker first

**Comment by insanelydeepak** [comment]
can you dump dmesg output from linux??

**Comment by hatf0** [comment]
[dmesg.txt](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2637503/dmesg.txt)

**Comment by insanelydeepak** [comment]
is your speaker working in linux??

**Comment by hatf0** [comment]
Yes, I have audio in Linux. The newest kext still has no audio (and for some reason Internal Speakers show up on the input instead of microphone), but it does immediately detect when the headphones have been plugged in.

**Comment by insanelydeepak** [comment]
do you have OSX 10.13 ??

**Comment by insanelydeepak** [comment]
try this 

[ALC299.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2643834/ALC299.zip)

**Comment by cepxuo** [comment]
> ping me when you available just last final step is left
Hi! I am back and ready to test.

**Comment by insanelydeepak** [comment]
Hi 

how stable it is ?

**Comment by insanelydeepak** [comment]
actually i forgot what command i use in v4 file, let me make it again might help better

**Comment by insanelydeepak** [comment]
try this one 

[alcfix.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2651785/alcfix.zip)

**Comment by cepxuo** [comment]
> try this one
> 
> [alcfix.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2651785/alcfix.zip)

V1 is a little louder, and V2 is a little more louder, so I can hear voice with system's maximum gain. However I have to speak directly to microphone's hole. Could you please add some gain?

**Comment by cepxuo** [comment]
> Hi
> 
> how stable it is ? anyother issue ?

It works stable, the only issue is headphones' microphone low sound.

**Comment by insanelydeepak** [comment]
let me see

**Comment by insanelydeepak** [comment]
try this , must remove AppleALC and use layout 14
[ALC299.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2659401/ALC299.zip)

**Comment by cepxuo** [comment]
Sorry for delay! V3 has the highest volume, but still noise is louder :(

**Comment by insanelydeepak** [comment]
share screenshot of input devices

**Comment by cepxuo** [comment]
![screen shot 2018-12-15 at 10 05 21](https://user-images.githubusercontent.com/8061075/50039281-7e21b500-0051-11e9-99a2-ef75da5c2e07.png)

**Comment by insanelydeepak** [comment]
>  but still noise is louder :(

Line-in or Mic ?? 
what about ambient noise reduction  it won't help ?

**Comment by cepxuo** [comment]
I think this is the problem. It is detected as Line-In, instead of Mic. AFAIK Line-In does not use builtin amplifier.
Ambient noise reduction does not help :(

**Comment by insanelydeepak** [comment]
let me check

**Comment by insanelydeepak** [comment]
try this 
[ALC299.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2688902/ALC299.zip)

**Comment by cepxuo** [comment]
Hi! I tried all 4 versions and all of them detect Mic as Line-In. All have almost same level of input signal (low), same as previous files. One more thing - readme says I need to setup Layout 13, but files work only with Layout 14.

**Comment by insanelydeepak** [comment]
ignore readme , i didnt updated it for below 10.14 use layout13 for 10.14 above layout 14 

LineIn = Mic External 
Mic = Mic internal 
now explain exact issue ?

**Comment by cepxuo** [comment]
So, since it did not work with layout 13 as readme said I used layout 14 as you said above. The issue is with recording sound from external Mic - it is very low. When I disable noise reduction I hear mostly noise and my voice is far-far away only the most loud sounds. This recording level is the same with all variants of AppleHDA.kext

**Comment by insanelydeepak** [comment]
i mean layout 13 is removed in Mojave so i changed my layout 14 but readMe is not updated yet 

try this 
[ALC299.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2705425/ALC299.zip)


also send your codec_dump again , i have 4-6 ALC299 codec so i forgot which one is yours

**Comment by cepxuo** [comment]
Hi! I am back again. Thank you for your patience and hard work on this project!!! And happy past holidays!!!

V1 has the best performance, but I have to apply patch from here to get input sound: https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/issues/15#issuecomment-448093124 

Now the volume is a little higher and I can hear some voice through the noise. The headphone's mic is still detected as Line-in

And attaches are the codec dumps:
[Codec_dumps.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2798655/Codec_dumps.zip)

**Comment by cepxuo** [comment]
No problem with "Linein" wording. 
If you could boost a little more mic volume it would be great! The version I use now (V1 with patch) has the highest volume comparing with others but it still lower than built-in mic.

**Comment by insanelydeepak** [comment]
> If you could boost a little more mic volume it would be great!

Line-In ? right

**Comment by cepxuo** [comment]
yes. the volume from external mic, which is displayed as line-in.

**Comment by insanelydeepak** [comment]
> The version I use now (V1 with patch) has the highest volume comparing

this one ,
send me your current patched file

**Comment by cepxuo** [comment]
Here it is:
[AppleHDA.kext.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2821550/AppleHDA.kext.zip)

**Comment by insanelydeepak** [comment]
try this one 
[AppleHDA.kext.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/2825189/AppleHDA.kext.zip)

**Comment by saeckl** [comment]
Could you help me as well?
I also have an ALC299
OS 10.13.6 Laptop Helios 500 PH517-51-99UR
[dump.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/3149464/dump.zip)

**Comment by insanelydeepak** [comment]
Hi 
sure let me check

**Comment by saeckl** [comment]
thx maybe this will help ..

Linux Codec Dump, and the built-in audio driver doesn’t detect the headphone jack connection properly. I had to use hda-jack-retask to indicate that your socket is disconnected from Ubuntu.

[codec_dump.txt](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/3158481/codec_dump.txt)
![override](https://user-images.githubusercontent.com/50339370/57393103-b07bac80-71c2-11e9-8cfb-3b07387e24bd.png)

**Comment by insanelydeepak** [comment]
Hi 
i was out of town from months so wasn't able to look into it 
do you still need help ? feel free to re-open again

**Comment by ZOUHIRZED** [comment]
hi I have ups 13 9380 alc299 (oc 10.15.7). im using alc/commader with layout id 22. .speaker /mic working but for the combo jack I'm using  [https://github.com/wdubaiyu/Hackintosh-Dell-XPS-9380/tree/master/ALCPlugFix](url)  to fix distortion
, but if use volume beyond 80% I get distortion back .is there any solution for this ?
[xps9380EffectiveNode.txt](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/5419324/xps9380EffectiveNode.txt)
[codec.0-2.txt](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/5419325/codec.0-2.txt)
